alembic/versions/fe7656c21eaa_add_remainder_of_database_models.py

```diff
@@ -0,0 +1,67 @@
+"""add remainder of database models
+
+Revision ID: fe7656c21eaa
+Revises: 5344fc3a3d3f
+Create Date: 2018-08-01 15:44:29.298131
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'fe7656c21eaa'
+down_revision = '5344fc3a3d3f'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        'sources',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('uuid', sa.String(length=36), nullable=False),
+        sa.Column('journalist_designation', sa.String(length=255),
+                  nullable=False),
+        sa.Column('is_flagged', sa.Boolean(), nullable=True),
+        sa.Column('public_key', sa.String(length=10000), nullable=True),
+        sa.Column('interaction_count', sa.Integer(), nullable=False),
+        sa.Column('is_starred', sa.Boolean(), nullable=True),
+        sa.Column('last_updated', sa.DateTime(), nullable=True),
+        sa.PrimaryKeyConstraint('id'),
+        sa.UniqueConstraint('uuid')
+    )
+    op.create_table(
+        'replies',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('source_id', sa.Integer(), nullable=True),
+        sa.Column('journalist_id', sa.Integer(), nullable=True),
+        sa.Column('filename', sa.String(length=255), nullable=False),
+        sa.Column('size', sa.Integer(), nullable=False),
+        sa.ForeignKeyConstraint(['journalist_id'], ['users.id'], ),
+        sa.ForeignKeyConstraint(['source_id'], ['sources.id'], ),
+        sa.PrimaryKeyConstraint('id')
+    )
+    op.create_table(
+        'submissions',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('uuid', sa.String(length=36), nullable=False),
+        sa.Column('filename', sa.String(length=255), nullable=False),
+        sa.Column('size', sa.Integer(), nullable=False),
+        sa.Column('is_downloaded', sa.Boolean(), nullable=True),
+        sa.Column('is_read', sa.Boolean(), nullable=True),
+        sa.Column('source_id', sa.Integer(), nullable=True),
```

Comment:
Do we want a date on submissions?

Do we want to keep other state on submissions, to indicate to other users stuff aside from "is_read"?

Reply:
> Do we want a date on submissions?

#13 applies here

> Do we want to keep other state on submissions, to indicate to other users stuff aside from "is_read"?

Thinking beyond the MVP for a second, we definitely do want to store additional per-submission and per-source data when we can store it on the server (encrypted by the clients) as backup, likely candidates are [custom tags](https://github.com/freedomofpress/securedrop/pull/1422), journalist assignments (something that has frequently come up in user research are the various hacky methods that organizations are using to manage which journalists are working on which SecureDrop stories, this is information that we should store in the workstations). What do you think - any other ideas we should keep in mind?